zerver/forms.py

```diff
@@ -210,8 +211,11 @@ def send_mail(self, subject_template_name, email_template_name,
         the mail through the functions in zerver.lib.send_email, to match
         how we send all other mail in the codebase.
         """
+        request = getattr(self, 'request')
         user = get_user_profile_by_email(to_email)
-        attempted_subdomain = get_subdomain(getattr(self, 'request'))
+        attempted_subdomain = get_subdomain(request)
+        if settings.EMAIL_RATE_LIMITING:
+            rate_limit_email(request, to_email, attempted_subdomain)
```

Comment:
> Over here I just wanted to stop someone from using Zulip to spam people with password reset emails.

I don't think this is effective for this purpose -- if we had a problem with that that we were trying to solve, we'd want to set a very different limit value than for attempting a password. E.g., if we limit login attempts to 1 per 10s for a given email, that will be very strong throttling for defeating online password cracking... but 1 reset email per 10s would be >8000/day and extremely spammy.

We might eventually find the same ratelimit infrastructure is useful for that potential spam issue too, but let's leave it aside for now.